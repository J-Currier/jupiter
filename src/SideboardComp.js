import React from 'react';

// import { ReactComponent as IconUp } from './images/kuba_arrow_button_set_3.svg'
// import { ReactComponent as IconLeft } from './images/kuba_arrow_button_set_1.svg'
// import { ReactComponent as IconRight } from './images/kuba_arrow_button_set_2.svg'
// import { ReactComponent as IconDown } from './images/kuba_arrow_button_set_4.svg'
import IconUp from './images/kuba_arrow_button_set_3.svg'
import IconLeft from './images/kuba_arrow_button_set_1.svg'
import IconRight from './images/kuba_arrow_button_set_2.svg'
import IconDown from './images/kuba_arrow_button_set_4.svg'
import IconUndo from './images/arrow-curved-blue.svg'

function Sideboard(props) {

    function HandleClick(e) {
        
        buttonInfo[e.target.name].fx();
    };
    const buttonInfo = {
        up: {icon: IconUp, fx: ()=>{console.log("up")}},
        down: {icon: IconDown, fx: ()=>{console.log("Down")}},
        left: {icon: IconLeft, fx: ()=>{console.log("<--")}},
        right: {icon: IconRight, fx: ()=>{console.log("Droit")}},
        undo: {icon: IconUndo, fx: ()=>{console.log("undo")}},
        redo: {icon: IconUndo, fx: ()=>{console.log("redo")}}
    };

    const buttons = [];
    for(let index in buttonInfo) {
        if (index==="up" || index==="down" || index==="left" || index==="right") {
            buttons.push(
                <img 
                    onClick={HandleClick} 
                    key={index} 
                    name={index} 
                    className="directional button" 
                    src= {buttonInfo[index].icon}
                ></img>
            );
        };
    };

    const timeButtons = [];
    for(let index in buttonInfo) {
        if (index==="undo" || index==="redo") {
            timeButtons.push(
                <img 
                    onClick={HandleClick} 
                    key={index} 
                    name={index} 
                    className="time button" 
                    src= {buttonInfo[index].icon}
                ></img>
            );
        };
    };


    return (
        <div>   
            {buttons}
            {timeButtons}
        </div>
    )
}

export default Sideboard;