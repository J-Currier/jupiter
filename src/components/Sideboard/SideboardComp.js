import React, { useState } from "react";
import "./Sideboard.css";
import Translation from "../translation/translationComp";
import Rotation from "../rotation/rotationComp";
import Reflection from "../reflection/reflectionComp";
import { ReactComponent as TranslationSvg } from "../../images/btnTrans.svg";
import { ReactComponent as RotationSvg } from "../../images/btnRot.svg";
import { ReactComponent as ReflectionSvg } from "../../images/btnReflect.svg";

function Sideboard(props) {
  const [tab, setTab] = useState(null);

  function handleClick(e) {
    if (e.currentTarget.id === "translation-button") {
      setTab("translation");
    } else if (e.currentTarget.id === "reflection-button") {
      setTab("reflection");
    } else if (e.currentTarget.id === "rotation-button") {
      setTab("rotation");
    }
  }

  return (
    <div id="sideboard">
      <nav>
        <button
          id="translation-button"
          className="tabBtn iconBtn"
          onClick={handleClick}
        >
          <TranslationSvg 
            alt="Translation" 
            tabIndex="0"
            className={(tab === "translation")? "activeTab": ""}
          />
        </button>
        <button
          id="rotation-button"
          className="tabBtn iconBtn"
          onClick={handleClick}
        >
          <RotationSvg 
            alt="Rotation" 
            tabIndex="0"
            className={(tab === "rotation")? "activeTab": ""}
          />
        </button>
        <button
          id="reflection-button"
          className="tabBtn iconBtn"
          onClick={handleClick}
        >
          <ReflectionSvg 
            alt="Reflection" 
            tabIndex="0"
            className={(tab === "reflection")? "activeTab": ""}
          />
        </button>
      </nav>
      {tab === "translation" && (
        <Translation
          translate={props.translate}
          translationFactor={props.translationFactor}
          handleChange={props.handleChange}
          addToStack={props.addToStack}
        />
      )}
      {tab === "rotation" && (
        <Rotation
          rotate={props.rotate}
          pivotPointx={props.pivotPointx}
          pivotPointy={props.pivotPointy}
          rotationMag={props.rotationMag}
          handleChange={props.handleChange}
          addToStack={props.addToStack}
        />
      )}
      {tab === "reflection" && (
        <Reflection
          reflect={props.reflect}
          lineOfReflection={props.lineOfReflection}
          handleChange={props.handleChange}
          addToStack={props.addToStack}
        />
      )}
    </div>
  );
}

export default Sideboard;
