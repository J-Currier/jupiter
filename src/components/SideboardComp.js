import React, {useState} from 'react';

// import { ReactComponent as IconUp } from './images/kuba_arrow_button_set_3.svg'
// import { ReactComponent as IconLeft } from './images/kuba_arrow_button_set_1.svg'
// import { ReactComponent as IconRight } from './images/kuba_arrow_button_set_2.svg'
// import { ReactComponent as IconDown } from './images/kuba_arrow_button_set_4.svg'
import IconUp from '../images/kuba_arrow_button_set_3.svg'
import IconLeft from '../images/kuba_arrow_button_set_1.svg'
import IconRight from '../images/kuba_arrow_button_set_2.svg'
import IconDown from '../images/kuba_arrow_button_set_4.svg'
import IconUndo from '../images/arrow-curved-blue.svg'
import './Sideboard.css'

function Sideboard(props) {

    const [step, setStep] = useState(0); // todo

    function HandleClick(e) {
        buttonInfo[e.target.name].fx(e)
    };
    const buttonInfo = {
        up: {icon: IconUp, fx: (e)=>{props.buttonFunction(e, 0, -100)}},
        left: {icon: IconLeft, fx: (e)=>{props.buttonFunction(e, -100, 0)}},
        right: {icon: IconRight, fx: (e)=>{props.buttonFunction(e, 100, 0)}},
        down: {icon: IconDown, fx: (e)=>{props.buttonFunction(e, 0, 100)}},
        undo: {icon: IconUndo, fx: setStep},
        redo: {icon: IconUndo, fx: setStep}
    };

    const buttons = [];
    for(let index in buttonInfo) {
        if (index==="up" || index==="down" || index==="left" || index==="right") {
            buttons.push(
                <img 
                    onClick={HandleClick} 
                    key={index} 
                    name={index} 
                    id={index}
                    className="directional button" 
                    src= {buttonInfo[index].icon}
                ></img>
            );
        };
    };

    // const timeButtons = [];
    // for(let index in buttonInfo) {
    //     if (index==="undo" || index==="redo") {
    //         timeButtons.push(
    //             <img 
    //                 onClick={HandleClick} 
    //                 key={index} 
    //                 name={index}
    //                 id={index} 
    //                 className="time button" 
    //                 src= {buttonInfo[index].icon}
    //             ></img>
    //         );
    //     };
    // };

    
    function Factor(props) {
        return(
            <input 
                type="number" 
                min='0' 
                max='18' 
                onChange={props.factorHandle} 
                value={props.moveFactor} 
                id="factorBox"
                key="factorKey"
                autoFocus= {true}
            />
        )
    }
    return (
        <div id="sideboard">   
            <Factor key='FactorKey'
                moveFactor={props.moveFactor}
                factorHandle={props.factorHandle}
            />
            {buttons}
            {/* {timeButtons} */}
        </div>
    )
}

export default Sideboard;