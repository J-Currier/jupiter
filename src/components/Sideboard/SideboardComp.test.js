import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";

import SideboardComp from "./SideboardComp";

describe("sideboard load", () => {
  let container = null;

  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  test("render", () => {
    act(() => {
      render(<SideboardComp />, container);
    });
  });
})

describe("sideboard tab clicks", () => {
  let container = null;
  let translate, rotate, reflect, handleChange, translationFactor, pivotPointx, pivotPointy, rotationMag, lineOfReflection;
  let translationBtn, reflectionBtn, rotationBtn;
  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);

    act(() => {
      render(
        <SideboardComp
        translate={translate}
        rotate={rotate}
        reflect={reflect}
        handleChange={handleChange}
        translationFactor={translationFactor}
        pivotPointx={pivotPointx}
        pivotPointy={pivotPointy}
        rotationMag ={rotationMag}
        lineOfReflection={lineOfReflection}
        />,
        container
      );
    });
    translationBtn = document.getElementById("translation-button");
    rotationBtn = document.getElementById("rotation-button");
    reflectionBtn = document.getElementById("reflection-button");
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  test("click translation", () => {
    // click translation
    act(() => {
      translationBtn.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });

  test("click rotation", () => {
    // click rotation
    act(() => {
      rotationBtn.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });

  test("click reflection", () => {
    // click reflection
    act(() => {
      reflectionBtn.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
