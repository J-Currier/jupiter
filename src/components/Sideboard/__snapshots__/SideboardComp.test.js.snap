// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sideboard tab clicks click reflection 1`] = `
"<div id=\\"sideboard\\">
  <nav><button id=\\"translation-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Translation\\" tabindex=\\"0\\" class=\\"\\">btnTrans.svg</svg></button><button id=\\"rotation-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Rotation\\" tabindex=\\"0\\" class=\\"\\">btnRot.svg</svg></button><button id=\\"reflection-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Reflection\\" tabindex=\\"0\\" class=\\"activeTab\\">btnReflect.svg</svg></button></nav>
  <div id=\\"reflection\\" class=\\"tab\\">
    <div class=\\"controls\\"><button id=\\"reflectXBtn\\" name=\\"reflectX\\" class=\\"controlBtn iconBtn\\"><svg alt=\\"X-reflection icon\\">btnReflectX.svg</svg></button><select id=\\"reflectDrop\\" name=\\"setLineOfReflection\\" class=\\"controlSelect\\">
        <option>9</option>
        <option>8</option>
        <option>7</option>
        <option>6</option>
        <option>5</option>
        <option>4</option>
        <option>3</option>
        <option>2</option>
        <option>1</option>
        <option>0</option>
        <option>-1</option>
        <option>-2</option>
        <option>-3</option>
        <option>-4</option>
        <option>-5</option>
        <option>-6</option>
        <option>-7</option>
        <option>-8</option>
        <option>-9</option>
      </select><button id=\\"reflectYBtn\\" name=\\"reflectY\\" class=\\"controlBtn iconBtn\\"><svg alt=\\"Y-reflection icon\\">btnReflectY.svg</svg></button></div>
  </div>
</div>"
`;

exports[`sideboard tab clicks click rotation 1`] = `
"<div id=\\"sideboard\\">
  <nav><button id=\\"translation-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Translation\\" tabindex=\\"0\\" class=\\"\\">btnTrans.svg</svg></button><button id=\\"rotation-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Rotation\\" tabindex=\\"0\\" class=\\"activeTab\\">btnRot.svg</svg></button><button id=\\"reflection-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Reflection\\" tabindex=\\"0\\" class=\\"\\">btnReflect.svg</svg></button></nav>
  <div id=\\"rotation\\" class=\\"tab\\">
    <div class=\\"controls\\">
      <div id=\\"pivotCoords\\"><input id=\\"setPivotPointx\\" class=\\"controlInputNum\\" type=\\"number\\" name=\\"setPivotPointx\\" value=\\"\\"><input id=\\"setPivotPointy\\" class=\\"controlInputNum\\" type=\\"number\\" name=\\"setPivotPointy\\" value=\\"\\"></div>
      <div id=\\"rotateDirection\\"><button id=\\"rotate-left\\" name=\\"rotate-left\\" class=\\"controlBtn iconBtn\\"><svg alt=\\"Counterclockwise Rotate Icon\\">btnRotCounter.svg</svg></button><select id=\\"rotateDrop\\" name=\\"setRotationMag\\" class=\\"controlSelect\\">
          <option>90°</option>
          <option>180°</option>
          <option>270°</option>
        </select><button id=\\"rotate-right\\" name=\\"rotate-right\\" class=\\"controlBtn iconBtn\\"><svg alt=\\"Clockwise Rotate Icon\\" class=\\"reflectXBtn\\">btnRotCounter.svg</svg></button></div>
    </div>
  </div>
</div>"
`;

exports[`sideboard tab clicks click translation 1`] = `
"<div id=\\"sideboard\\">
  <nav><button id=\\"translation-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Translation\\" tabindex=\\"0\\" class=\\"activeTab\\">btnTrans.svg</svg></button><button id=\\"rotation-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Rotation\\" tabindex=\\"0\\" class=\\"\\">btnRot.svg</svg></button><button id=\\"reflection-button\\" class=\\"tabBtn iconBtn\\"><svg alt=\\"Reflection\\" tabindex=\\"0\\" class=\\"\\">btnReflect.svg</svg></button></nav>
  <div id=\\"translation\\" class=\\"tab\\">
    <div class=\\"controls\\"><input name=\\"setTranslationFactor\\" type=\\"number\\" min=\\"0\\" max=\\"18\\" id=\\"factorBox\\" class=\\"controlInputNum\\" value=\\"\\"><button name=\\"up\\" alt=\\"up button\\" id=\\"up\\" class=\\"controlBtn iconBtn\\"><svg class=\\"upBtn\\">btnTransArrow.svg</svg></button><button name=\\"left\\" alt=\\"left button\\" id=\\"left\\" class=\\"controlBtn iconBtn\\"><svg class=\\"leftBtn\\">btnTransArrow.svg</svg></button><button name=\\"right\\" alt=\\"right button\\" id=\\"right\\" class=\\"controlBtn iconBtn\\"><svg class=\\"rightBtn\\">btnTransArrow.svg</svg></button><button name=\\"down\\" alt=\\"down button\\" id=\\"down\\" class=\\"controlBtn iconBtn\\"><svg class=\\"downBtn\\">btnTransArrow.svg</svg></button></div>
  </div>
</div>"
`;
